tools/autograd/templates/Functions.cpp

```diff
@@ -1970,68 +1970,125 @@ Tensor symeig_backward(const std::vector<torch::autograd::Variable> &grads, cons
+    Tensor solve_soln_1;
+    std::tie(solve_soln_1, std::ignore) = at::triangular_solve(rhs_solve_1.transpose(-2, -1), R,
                                                              /*upper=*/true, /*transpose=*/false,
                                                              /*unitriangular=*/false);
-  Tensor grad_A;
-  if (grad_R.defined()) {
-    grad_A = at::matmul(Q, solve_soln_1.transpose(-2, -1) + grad_R);
-  } else {
-    grad_A = at::matmul(Q, solve_soln_1.transpose(-2, -1));
-  }
-
-  // Successive computations involve computation of QQ^{T} which is identity when A is square
-  if (self.size(-1) != self.size(-2)) {
-    Tensor rhs_solve_2;
-    // We use the same trick from above for this computation
-    if (grad_Q.defined()) {
-      rhs_solve_2 = grad_Q - at::matmul(Q, Q_term);
+    Tensor grad_A;
+    if (grad_R.defined()) {
+      grad_A = at::matmul(Q, solve_soln_1.transpose(-2, -1) + grad_R);
     } else {
-      rhs_solve_2 = -at::matmul(Q, Q_term);
+      grad_A = at::matmul(Q, solve_soln_1.transpose(-2, -1));
     }
-    Tensor solve_soln_2;
-    std::tie(solve_soln_2, std::ignore) = at::triangular_solve(rhs_solve_2.transpose(-2, -1), R,
+
+    // Successive computations involve computation of QQ^{T} which is identity when A is square
+    if (A.size(-1) != A.size(-2)) {
+      Tensor rhs_solve_2;
+      // We use the same trick from above for this computation
+      if (grad_Q.defined()) {
+        rhs_solve_2 = grad_Q - at::matmul(Q, Q_term);
+      } else {
+        rhs_solve_2 = -at::matmul(Q, Q_term);
+      }
+      Tensor solve_soln_2;
+      std::tie(solve_soln_2, std::ignore) = at::triangular_solve(rhs_solve_2.transpose(-2, -1), R,
                                                                /*upper=*/true, /*transpose=*/false,
                                                                /*unitriangular=*/false);
-    grad_A.add_(solve_soln_2.transpose(-2, -1));
+      grad_A.add_(solve_soln_2.transpose(-2, -1));
+    }
+    return grad_A;
+  };
+
+  //NOTE: grad of complete QR cannot exist b/c inverse not exist/unique
+  TORCH_CHECK(((r.size(-2) <= r.size(-1) && (!some)) || some),
+		     "The derivative is not implemented when nrows > ncols and complete QR. ");
+
+  auto grad_Q = grads[0];
+  auto grad_R = grads[1];
+
+  auto m = self.size(-2);
+  auto n = self.size(-1);
+
+ if (m >= n) {
+    return square_deep_case_backward(grad_Q, grad_R, self, q, r);
+  } else {
+    // For wide (m > n) input matrices A,  partition A = [X|Y] and R = [U|V]
+    // X and U are square full rank matrices. Also partition
+    // grad_R = [grad_U, grad_V] and grad_A = [grad_X | grad_Y].
+    // To obtain grad_X we reuse the gradient formula from the square case.
+    // grad_Y  Then concatenate grad_A = [grad_X, grad_Y].
+    // Formulae: grad_X = square_case_grad(grad_Q_prime, grad_U, Q, U),
+    // where grad_q_prime = grad_Q + Y @ grad_V.T
+    // and grad_Y = Q @ grad_V.
```

Comment:
This comment is using the python `@` operator for the (Euclidean) inner product. If this is confusing as written I can replace with the equivalent operator for the `Tensor` class.